txhttptrace.py

```diff
@@ -80,7 +82,10 @@ def wrapper(ctx, request, *args, **kwargs):
         def do_finally(param):
             ctx['took'] = time.time() - start
             if 'error' in ctx:
-                _logger.error('request failed', **ctx)
+                log_error = _logger.error
+                if _log_errors_as_warnings:
+                    log_error = _logger.warn
+                log_error('request failed', **ctx)
```

Comment:
This would probably be more legible using if-else rather than renaming the logging function. How about:

```python
msg = 'request failed'
if _log_errors_as_warnings:
    _logger.warn(msg, **ctx)
else:
    _logger.error(msg, **ctx)
```